### DISCUSSION TITLE
Config File Value for downsample_mode not properly loading in on SMuRF Power up

### DISCUSSION THREAD

**Discussion by samdayweiss** [simonsobs/sodetlib#402] [General]
Hi all,

My understanding is that the config file variable `downsample_mode` must be set to `external` in order for co-sampling across different slots to be done. I have changed all of the smurf slot configuration files to this, however upon smurf shutdown and restart, the variable is never correctly loaded in (I always have to change it manually using `s.set_downsample_mode('external'))`.

Has anyone seen this kind of configuration loading issue before?

**Comment by jlashner** [comment]
Can you describe how you're checking that the `downsample_mode` is not being set correctly? This is not set in the pysmurf object until starting a streaming session: https://github.com/simonsobs/sodetlib/blob/1fa5ba54842d09ba588b5afb022056dc71bbde3c/sodetlib/stream.py#L201-L206

So its quite possible when you start up for the first time, `cfg.dev.exp['downsample_mode']` is correct, but `S.get_downsample_mode()` won't return the right value.

**Reply by samdayweiss** [reply]
Hi Jack, thanks this is super helpful, I was querying via `S.get_downsample_mode()`, so if stream loads it in after the fact, it may very well be fine!
